### ISSUE TITLE
Translate 2019 Content into Chinese

### ISSUE THREAD

**Issue by bazzadp** [HTTPArchive/almanac.httparchive.org#1067]
These are the chapters to be translated, and they exist in the [content directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content):
- [ ] accessibility - 
- [ ] caching - 
- [ ] cdn - 
- [ ] cms - 
- [ ] compression - 
- [ ] css - 
- [ ] ecommerce - 
- [ ] fonts - 
- [ ] http2 - 
- [ ] javascript - 
- [ ] markup - 
- [ ] media - 
- [ ] mobile-web - 
- [ ] page-weight - 
- [ ] performance - 
- [ ] pwa - 
- [ ] resource-hints - 
- [ ] security - 
- [ ] seo - 
- [ ] third-parties - 

Please include "Makes progress on #926" in all pull requests so a link is created from the PR to this issue.

Additionally the following pages need translated too in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [ ] base.html
- [ ] base_chapter.html
- [ ] contributors.html
- [ ] error.html
- [ ] index.html
- [ ] featured_chapters.html
- [ ] methodology.html 
- [ ] table_of_contents.html 
- [ ] accessibility_statement.html 
- [ ] base_ebook.html

There is no need to translate the chapters HTML pages in `src/templates/en/2019/chapters/` nor the `src/templates/en/2019/ebook.html` file as they are generated off the markdown combined with the above templates.

---

Common notes for writing consistency are here: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide. Feel free to edit that and/or add Chinese-specific extras by editing this comment.

Chinese specific extra advice:

*

**Comment by chengxicn** [comment]
start  translating language specific templates directory.

**Comment by bazzadp** [comment]
We found it best to put your name besides the parts you are doing on the first comment to tell others not to look at that one. I’ve done that with `base.html` as an example. Feel free to remove it from there and/or add your name to others you are working on so @AppleTsai knows to work on other ones.

**Comment by AbbyTsai** [comment]
Thanks your warm markers and tips.

**Comment by chengxicn** [comment]
Submitted first PR,but there are some errors(from 311) in accessibility.md check https://github.com/HTTPArchive/almanac.httparchive.org/pull/1072/checks ,  but I only translated  front content，not sure why there are hard tabs can't pass the check yet. So have to delete accessibility.md  and submitted a new PR again.

**Comment by rviscomi** [comment]
@chengxicn @AbbyTsai I'd like to post something from the HTTP Archive Twitter account to promote this translation effort and hopefully find more translators to help with it. Could someone translate this text for the tweet?

> Do you speak Chinese? Help us translate the Web Almanac as part of our effort to make it globally accessible to the web community!

Thanks for your help!

**Comment by AbbyTsai** [comment]
徵的是會翻譯中文的你,歡迎加入我們Web Almanac網絡社群.讓我們共同攜手遨遊無邊際的語言世界.
@chengxicn   it's open to your comment.
I'll try to leave the message to network, hoping to pull more helper in translation.

**Comment by chengxicn** [comment]
> @chengxicn @AbbyTsai I'd like to post something from the HTTP Archive Twitter account to promote this translation effort and hopefully find more translators to help with it. Could someone translate this text for the tweet?
> 
> > Do you speak Chinese? Help us translate the Web Almanac as part of our effort to make it globally accessible to the web community!
> 
> Thanks for your help!

您会说中文吗? 来帮助我们翻译Web Almanac 网络年鉴吧！让我们一起努力消除障碍，让全球Web社区自由访问！

@rviscomi  As I described before https://github.com/HTTPArchive/almanac.httparchive.org/pull/1073#discussion_r459603706    , mine and @AbbyTsai 's Chinese are very different systems, it's very diffenrent.  I'm working on Simplified Chinese and @AbbyTsai is working on Traditional Chinese, So we may need 2 groups of translators.

**Comment by rviscomi** [comment]
Thanks @AbbyTsai @chengxicn! I'll post two separate tweets using each dialect.

@bazzadp it'd be good to split this issue into Simplified and Traditional so we can disambiguate for translators. I've renamed #1073 to clarify the dialect. For that PR I'd recommend completing the English translations into Simplified Chinese and waiting for a Simplified Chinese speaker to review it, or merge the PR if we can't find anyone soon. @AbbyTsai would you be able to work on a Traditional Chinese translation in parallel?

With two separate issues I'll be able to link the tweets to their respective dialects.

**Comment by bazzadp** [comment]
That's done - #1106 

Only question remains is do we use the `zn` folder and URL structure for one or the other?
Or have `zh-cn` and `zh-hk`? In which case @chengxicn can you update your PR?

**Comment by chengxicn** [comment]
I'm not sure if the folders have to comply with the ISO standard?  could I name them 'cn-simp' and 'cn-trad' ?

**Comment by bazzadp** [comment]
Technically they don't but would prefer to use something that's not completely unusal!

According to this https://stackoverflow.com/a/4894634/2144578 `zh_CN` and `zh_TW` seem to be the most common. Don't love introducing case into URLs though but can always revert to lowercase if it causes problems.

But willing to take your steer @chengxicn and @AbbyTsai since I'm sure you browser a lot more Chinese internet sites than me! 😀

**Comment by chengxicn** [comment]
Updated to zh_CN.

**Comment by AbbyTsai** [comment]
@rviscomi @bazzadp I wish I could have triple hands to deal with both role of analyst and parallel translation, however, things don't go the way. After that schedule, I'll love to do translating in 2020 version. Thanks.
@chengxicn Happy to co-work together. Hoping to partner in the next project.

**Comment by rviscomi** [comment]
Thank you @AbbyTsai I hope we can find more translators to help with that. I've just started posting the language-specific tweets:

- Chinese (Simplified): https://twitter.com/HTTPArchive/status/1287831105652097024
- Chinese (Traditional): to be posted later this week, I'll edit this when it's live

**Comment by bazzadp** [comment]
@chengxicn I went ahead and merged #1073 but spotted some minor issues when doing final testing on it Not enough to hold up the merge as it was quite a big PR to leave out there, but if you could review and fix any of these that need fixing:
- [ ] - Methodology tables headings (lines 177 - 129, 184 - 186, 190, 200, 212, 218, 265, 266, 271, 275, 279,  - should it be translated?
- [ ] - Base.html (line 1240) - "Generated with PrinceXML"

**Comment by limhenry** [comment]
@chengxicn I am willing to translate the some of the chapters to zh-CN 🙋  Let me know if you need any help.

**Comment by chengxicn** [comment]
> @chengxicn I am willing to translate the some of the chapters to zh-CN 🙋 Let me know if you need any help.

Hi @limhenry, Great!  The work can be faster! I'm working on simplified Chinese accessibility.md chapter translation, please put your name besides the chapter you are doing.

**Comment by bazzadp** [comment]
Great news @limhenry - just sent you an invite to the team. If you could review #1073 that was already merged that would be good, and if you spot any errors then open a PR for that.

**Comment by chengxicn** [comment]
> @chengxicn I went ahead and merged #1073 but spotted some minor issues when doing final testing on it Not enough to hold up the merge as it was quite a big PR to leave out there, but if you could review and fix any of these that need fixing:
> 
> * [ ]  - Methodology tables headings (lines 177 - 129, 184 - 186, 190, 200, 212, 218, 265, 266, 271, 275, 279,  - should it be translated?
> * [ ]  - Base.html (line 1240) - "Generated with PrinceXML"

I've fixed this 2 issues, will open PR with accessibility.md chapter together.

**Comment by rviscomi** [comment]
Great work @chengxicn! Excited to the the base translations live! https://almanac.httparchive.org/zh-CN/2019/

**Comment by Zuckjet** [comment]
I can help with chinese translation of 2020,  especially about pwa and web security, since I have done these things in my work project this year

**Comment by bazzadp** [comment]
Excellent! Edit the first post to stick your name beside a chapter and away you go!

Original english is in `src/content/en/2020/` and should be translated and pull requested into `src/content/zh-CN/2020`.

Any questions feel free to ask!

**Comment by Zuckjet** [comment]
@bazzadp  I just edit the first comment under this issue like "performance- @Zuckjet". Does this mean I will take over the performance post for translation ?

**Comment by bazzadp** [comment]
Correct. But let’s untick it until it’s actually finish and PR accepted.

**Comment by Zuckjet** [comment]
> Correct. But let’s untick it until it’s actually translated and PR accepted.

Thanks for helping untick the list item, I will begin translate in these days

**Comment by bazzadp** [comment]
One thing to note is that for accessibility reasons whenever we are leaving a phrase in the original English we like to add `lang="en"` attributes to that text (wrapping in a `<span`> tag if necessary).

Like this example in the Capabilities translation:
https://github.com/HTTPArchive/almanac.httparchive.org/blob/4883d3038ab4aaeb5b408832c9151c5d05d39ce5/src/content/zh-CN/2020/capabilities.md#L27

**Comment by Zuckjet** [comment]
the seo chapter of 2020 isn't be marked with unedited flag, is it ready for translate ?

**Comment by bazzadp** [comment]
Yes it is. Updated the first comment with the latest list.

**Comment by Zuckjet** [comment]
> Yes it is. Updated the first comment with the latest list.

I remember have checked performance, since have finished translation. but seems someone uncheck it again, anything should i do about that chapter?

**Comment by bazzadp** [comment]
Sorry my fault. Got it confused with Traditional Chinese which hasn’t been merged yet. I have re-ticked it.

**Comment by tunetheweb** [comment]
We added a small note to our Accessibility Statement about colour contrast issues:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/14f85122c88d14ee1f8dd0474479424338766625/src/templates/zh-CN/accessibility_statement.html#L69-L71

Would be great if someone could translate this (@chengxicn ?). Very small and easy!

**Comment by EryouHao** [comment]
@tunetheweb Hi, has the translation program for 2022 started yet? I'm interested in translating the simplified Chinese
